### PR TITLE
Loop error fix and backward compability support

### PR DESCRIPTION
**Pull request by brostosjoined**
https://github.com/bombsquad-community/plugin-manager/issues/129#issuecomment-1742072208
image

fixes this and campability issues brought about by the previous pull request not tested though

### CODE REVIEW

**Comment by brostosjoined** [comment]
I dont know why its bringing api 7 mods??

File: plugin_manager.py [hunk 1]
```diff
@@ -2073,7 +2072,7 @@ async def draw_ui(self):
                        size=(0, 0),
                        h_align='center',
                        v_align='center',
-                       text=f'API Version: {babase.app.api_version}',
+                       text=f'API Version: {babase.app.api_version if build_number < 21282 else babase.app.env.api_version}',
```
Comment (rikkolovescats) [inline plugin_manager.py:2075]: this'll introduce lots of code changes that'll become legacy code in a few months (adding on to maintainence burden), let's add a compatibility layer instead, something like this:

```python
import babase
from baenv import TARGET_BALLISTICA_BUILD

if TARGET_BALLISTICA_BUILD < 21282:
    babase.app.env.api_version = babase.app.api_version
    babase.app.env.debug_build = babase.app.api_version.debug_build
    

# Now let's use `babase.app.env.*` everywhere else
# in code irrespective of whatever build version user is on.
```
then once a couple months have passed and most ppl are on newer versions, we can then simply nuke this compatibility layer code all at once easily.

File: plugin_manager.py [hunk 2]
```diff
@@ -808,7 +810,7 @@ def latest_compatible_version(self):
                     break
         if self._latest_compatible_version is None:
             raise NoCompatibleVersion(
-                f"{self.name} has no version compatible with API {babase.app.api_version}."
+                f"{self.name} has no version compatible with API {babase.app.api_version if build_number < 21282 else babase.app.env.api_version}."
```
Comment (rikkolovescats) [inline plugin_manager.py:813]: this'll introduce lots of code changes that'll become legacy code in a few months (adding on to maintainence burden), let's add a compatibility layer instead, something like this:

```python
import babase
from baenv import TARGET_BALLISTICA_BUILD

if TARGET_BALLISTICA_BUILD < 21282:
    babase.app.env.api_version = babase.app.api_version
    babase.app.env.debug_build = babase.app.debug_build
    

# Now let's use `babase.app.env.*` everywhere else
# in code irrespective of whatever build version user is on.
```
then once a couple months have passed and most ppl are on newer versions, we can then simply nuke this compatibility layer code all at once easily.

**Comment by rikkolovescats** [comment]
deprecated `babase.app` attributes are mentioned here:
https://github.com/efroemling/ballistica/blob/master/CHANGELOG.md#1727-build-21282-api-8-2023-08-30

> - Following up on the above two entries, several attributes under `babase.app`
  have been relocated to `babase.app.env` and the originals have been given
  deprecation warnings and will disappear sometime soon. This includes
  `build_number`, `device_name`, `config_file_path`, `version`, `debug_build`,
  `test_build`, `data_directory`, `python_directory_user`,
  `python_directory_app`, `python_directory_app_site`, `api_version`, `on_tv`,
  `vr_mode`, `toolbar_test`, `arcade_mode`, `headless_mode`, `demo_mode`, and
  `protocol_version`.

File: plugin_manager.py (hunk 2 above)
Comment (brostosjoined) [reply plugin_manager.py:813]: You can do those changes my laptop is dead

Comment (rikkolovescats) [reply plugin_manager.py:813]: ow, don't worry i'll take this battle up, get well soon lappy!

**Pull request merged**